test/cql-pytest/test_describe.py

```diff
@@ -447,6 +447,83 @@ def test_desc_index(cql, test_keyspace):
         
         assert f"CREATE INDEX named_index ON {tbl}{maybe_space}(c)" in c_desc
 
+def test_desc_index_on_collections(cql, test_keyspace):
+    # In this test, all assertions are in form of
+    # `assert create_stmt in desc or create_stmt.replace("(", " (", 1) in desc`
+    # This is because Scylla and Cassandra have slightly different (whitespace-wise)
+    # output of `DESC INDEX`. 
+    #   Scylla:     `CREATE INDEX ON table_name(column_name)`
+    #   Cassandra:  `CREATE INDEX ON table_name (column_name)`
+    
+    with new_test_table(cql, test_keyspace, "a int primary key, b int, c frozen<set<int>>, d map<int, int>, e list<int>, f set<int>, g list<int>, h set<int>, akeysb set<int>") as tbl:
+        indexes = [
+            ("idx_b",           f"CREATE INDEX idx_b ON {tbl}(b)"),
+            ("idx_c",           f"CREATE INDEX idx_c ON {tbl}(full(c))"),
+            ("idx_d_entries",   f"CREATE INDEX idx_d_entries ON {tbl}(entries(d))"),
+            ("idx_d_keys",      f"CREATE INDEX idx_d_keys ON {tbl}(keys(d))"),
+            ("idx_d_values",    f"CREATE INDEX idx_d_values ON {tbl}(values(d))"),
+            ("idx_e",           f"CREATE INDEX idx_e ON {tbl}(values(e))"),
+            ("idx_f",           f"CREATE INDEX idx_f ON {tbl}(values(f))"),
+            ("idx_akeysb",      f"CREATE INDEX idx_akeysb ON {tbl}(akeysb)"),
+        ]
+
+        for name, create_stmt in indexes:
+            cql.execute(create_stmt)
+            desc = cql.execute(f"DESC INDEX {test_keyspace}.{name}").one().create_statement
+            assert create_stmt in desc or create_stmt.replace("(", " (", 1)
+        
+        # When an index is created on list column with `CREATE INDEX ON tbl(list_column)`,
+        # its target is `values(list_column)` and describe will print creat statement with that
+        # values() function added.
+        cql.execute(f"CREATE INDEX idx_g ON {tbl}(g)")
+        idx_g_desc = cql.execute(f"DESC INDEX {test_keyspace}.idx_g").one().create_statement
+        idx_g_create_stmt = f"CREATE INDEX idx_g ON {tbl}(values(g))"
+        assert idx_g_create_stmt in idx_g_desc or idx_g_create_stmt.replace("(", " (", 1)
```

Comment:
I'm afraid `assert x in y or z` does not do what you think it does - to be fair, I'm not 100% sure what was your intention, but the way it works looks wrong. It's parsed as `assert (x in y) or z`. For example, this code does not trigger an assert:

```python
assert 0 in [1] or [2]
```

Python has a concept of "truthiness" and some values can be interpreted as booleans even though they aren't. In your case for example, non-empty strings (like `idx_g_create_stmt.replace("(", " (", 1)`) are always truthy. I'm afraid this assertion will always pass because the string at the end will be non-empty.

Was this supposed to be

```python
assert idx_g_create_stmt in idx_g_desc \
        or idx_g_create_stmt.replace("(", " (", 1) in idx_g_desc
```

?

Lots of assertions in the tests that you introduce have this problem.

Reply:
I recommend introducing a helper "normalize" function which gets rid of the space before the parenthesis (or adds it, but only if there was no space there in the first place) and then use the `in` operator on those.

Reply:
Yes, it wasn't intentional. I've simply made a mistake.